jit/core/jit_cleaner.py

```diff
@@ -0,0 +1,160 @@
+import json
+from datetime import datetime
+
+import google.auth
+import googleapiclient.discovery
+from google.api_core import retry
+from google.cloud import pubsub_v1
+from google.oauth2 import service_account  # type: ignore
+
+from jit.utils import config
+from jit.utils import constant
+from jit.utils.logger import jit_logger
+
+
+def modify_policy_remove_member(conf: config.JitConfig, target_project: str,
+  role: str, member: str,
+  start: datetime, end: datetime):
+    """Removes a  member from a role binding."""
+    jit_logger.info(
+      "modify_policy_remove_member. config=%s, target_project=%s, role=%s, member=%s, start=%s, end=%s",
+      conf, target_project, role, member, start, end
+    )
+    credentials, _ = google.auth.default()
+    service = googleapiclient.discovery.build(
+      "cloudresourcemanager", "v1", credentials=credentials
+    )
+    policy = (
+      service.projects()
+      .getIamPolicy(
+        resource=target_project,
+        body={"options": {"requestedPolicyVersion": 3}},
+      )
+      .execute()
+    )
+    jit_logger.debug(
+      "retrived policy. policy=%s, project=%s", policy, target_project
+    )
+    expression = f"(request.time >= timestamp(\"{start}\") && request.time < timestamp(\"{end}\"))"
+    if len(policy.get("bindings", [])) == 0:
+        jit_logger.error("project binding empty. ")
+        return
+    binding_count = 0
+    for b in policy["bindings"]:
+        if (
+          role == b["role"] and
+          member in b["members"] and
+          expression == b.get("condition", {}).get("expression", "")
```

Comment:
Given the title is static "JIT access activation" we should use that to ensure that the binding was created by JIT, in addition to the expression.

Reply:
thx for pointing out. changing it.

Reply:
done.